File: Trees/1. Invert Binary Tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	var temp = root.Left
	root.Left = invertTree(root.Right)
	root.Right = invertTree(temp)
	return root
}

File: Trees/1. Invert Binary Tree/solution.scala
/** Definition for a binary tree node. class TreeNode(_value: Int = 0, _left:
  * TreeNode = null, _right: TreeNode = null) { var value: Int = _value var
  * left: TreeNode = _left var right: TreeNode = _right }
  */
object Solution {
  def invertTree(root: TreeNode): TreeNode = {
    if (root == null) return null
    val temp = root.right
    root.right = invertTree(root.left)
    root.left = invertTree(temp)
    return root
  }
}
